GVFS/GVFS.Service/GvfsService.cs

```diff
@@ -348,19 +350,42 @@ private void LogExceptionAndExit(Exception e, string method)
             Environment.Exit((int)ReturnCode.GenericError);
         }
 
-        private void EnableAccessToAuthenticatedUsers(string rootDirectory)
+        private void CreateAndConfigureProgramDataDirectories()
         {
-            // GVFS Config is written to a temporary file and then renamed to its final destination.
-            // For this rename operation to succeed, user needs to have delete permission on the
-            // destination file, in case it is pre-existing. If the pre-existing file was created
-            // by a different user, then the delete will fail.
-            // Reference: https://stackoverflow.com/questions/22107812/privileges-owner-issue-when-writing-in-c-programdata
-            // This work around allows safe write to succeed in C:\ProgramData directory.
-
-            DirectorySecurity security = Directory.GetAccessControl(Path.GetDirectoryName(this.serviceDataLocation));
-            SecurityIdentifier authenticatedUsers = new SecurityIdentifier(WellKnownSidType.AuthenticatedUserSid, null);
-            security.AddAccessRule(new FileSystemAccessRule(authenticatedUsers, FileSystemRights.FullControl, AccessControlType.Allow));
-            Directory.SetAccessControl(Path.GetDirectoryName(this.serviceDataLocation), security);
+            this.serviceDataLocation = Paths.GetServiceDataRoot(this.serviceName);
+            string serviceDataRootPath = Path.GetDirectoryName(this.serviceDataLocation);
+
+            DirectorySecurity programDataSecurity = new DirectorySecurity();
+
+            // Protect the access rules from inheritance
+            programDataSecurity.SetAccessRuleProtection(isProtected: true, preserveInheritance: false);
+
+            // Everyone gets read access
+            SecurityIdentifier authenticatedUsers = new SecurityIdentifier(WellKnownSidType.WorldSid, null);

```

Comment:
Is World Sid what we want here? Given the name is `authenticatedUsers`, I was expecting the `AuthenticatedUser` Sid?

Reply:
Good point, I missed renaming the variable.  I actually switched from World to BuiltInUsers which are all users on the machine (I took a look at what other applications are doing that that seems to be pretty common).